### ISSUE TITLE
Support new bootstrap method in Flutter 3.22

### ISSUE THREAD

**Issue by EduardoYamauchi** [auth0/auth0-flutter#464]
### Checklist

- [X] The issue can be reproduced in the [auth0_flutter sample app](https://github.com/auth0-samples/auth0-flutter-samples/tree/main/sample) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/auth0-flutter/tree/main/auth0_flutter#readme), [Examples](https://github.com/auth0/auth0-flutter/blob/main/auth0_flutter/EXAMPLES.md), and [FAQ](https://github.com/auth0/auth0-flutter/blob/main/auth0_flutter/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://pub.dev/documentation/auth0_flutter/latest/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-flutter/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Description:

With the release of Flutter 3.22, the method for bootstrapping has changed, as outlined in the official documentation: [Flutter Web Bootstrapping](https://docs.flutter.dev/platform-integration/web/bootstrapping). As a result, the Auth0 integration no longer works with this new method.

Issue:

	• The previous bootstrap method is now deprecated and will soon stop functioning entirely.
	• The Auth0 package needs to be updated to support the new bootstrapping method introduced in Flutter 3.22.

Request:

Please provide support for the new bootstrap method in Flutter 3.22. This is crucial to ensure continued compatibility and functionality of Auth0 with Flutter web applications.

Thank you for your attention to this matter.

### Reproduction

1 - Follow the guide to update the bootstrapping in: https://docs.flutter.dev/platform-integration/web/bootstrapping
2 - Try to build an web application with auth0_flutter: 1.7.2

### auth0_flutter version

1.7.2

### Flutter version

3.22

### Platform

Web

**Comment by amangsingh** [comment]
I'm facing the same issue. Auth0 becomes completely unusable for web and is critical for my workflow

**Comment by Solano-Furlan** [comment]
Any updates on this?

**Comment by desusai7** [comment]
Hi @EduardoYamauchi, @amangsingh, @Solano-Furlan, 

Apologies for the delay in looking into this, 

I've just tried using the flutter sdk version 1.7.2 with the flutter versions 3.22.0 & 3.24.0 in both debug, release modes and everything is working as expected, can anyone of you help us with a reproducible example or steps to reproduce ?

**Comment by EduardoYamauchi** [comment]
I tested again using:

-  auth0_flutter: 1.7.2
-  Flutter (Channel stable, 3.24.2, on macOS 14.6.1 23G93 darwin-arm64).

And everything's is working as expected. I'm closing this issue, if you still having some issue related with the new bootstrap please send me a message to re-open.

Thank you.

**Comment by EduardoYamauchi** [comment]
I’m reopening this issue due to the following error:

```
TypeError: Cannot read properties of undefined (reading 'Auth0Client')
packages/auth0_flutter/src/web/auth0_flutter_plugin_real.dart 30:25  initialize
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54   runBody
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 127:5   _async
packages/auth0_flutter/src/web/auth0_flutter_plugin_real.dart 27:26  initialize
packages/auth0_flutter/auth0_flutter_web.dart 56:44                  onLoad
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54   runBody
```

This error occurs randomly only when switching to the **new bootstrap mode.**

Steps to reproduce (on web):

1. Add the required line for the web version in the index.html (following the documentation):

`<script src="https://cdn.auth0.com/js/auth0-spa-js/2.0/auth0-spa-js.production.js" defer></script>`
		
<img width="814" alt="Screenshot 2024-09-16 at 15 49 46" src="https://github.com/user-attachments/assets/063a75d1-94a2-4425-871a-f0d0f7d3c1cf">

2. Run the following lines:

```dart	
void main() async {
  var auth0Client = Auth0Client();
  await auth0Client.auth0Web!.onLoad();
}
```

I have been patiently waiting for the resolution of the [WASM issue](https://github.com/auth0/auth0-flutter/issues/463), which has now been open for 4 months since June. This is particularly important as it relates to performance and security of our applications.

**Comment by sw1jari** [comment]
simply removing `defer` resolves it for me

```html
<script src="https://cdn.auth0.com/js/auth0-spa-js/2.0/auth0-spa-js.production.js"></script>
```

**Comment by Widcket** [comment]
This is now out in [v1.9.0](https://github.com/auth0/auth0-flutter/releases/tag/af-v1.9.0).

**Comment by YMandarin** [comment]
Any update on this? I can only use this package either with the old bootstrapping (like the example app is using) or by removing defer on the spa-js

you should update the example to the new bootstrapper

**Comment by pmathew92** [comment]
Hi @YMandarin , Have you tried the v1.9.0 version of the Auth0 flutter SDK ?

**Comment by pmathew92** [comment]
@YMandarin , Yes `1.12.0` is he latest. Please confirm if you are observing the issue in the latest version also . We will take a look into this
